renderdoc/data/glsl/mesh.comp

```diff
@@ -40,37 +40,87 @@ layout(binding = 3, std140) buffer pickresult_buffer
 
 layout (local_size_x = 128, local_size_y = 1, local_size_z = 1) in;
 
+bool TriangleRayIntersect(vec3 A, vec3 B, vec3 C, 
+	vec3 RayPosition, vec3 RayDirection, out vec3 HitPosition)
+{
+	bool Result = false;
+
+	vec3 v0v1 = B - A;
+	vec3 v0v2 = C - A;
+	vec3 pvec = cross(RayDirection, v0v2);
+	float det = dot(v0v1, pvec);
+
+	// if the determinant is negative the triangle is backfacing
+	// if the determinant is close to 0, the ray misses the triangle
+	if (abs(det) > 0.000001f)
+	{
+		float invDet = 1 / det;
+
+		vec3 tvec = RayPosition - A;
+		vec3 qvec = cross(tvec, v0v1);
+		float u = dot(tvec, pvec) * invDet;
+		float v = dot(RayDirection, qvec) * invDet;
+
+		if (u > 0 && u < 1 &&
+			v > 0 && u + v < 1)
+		{
+			float t = dot(v0v2, qvec) * invDet;
+			HitPosition = RayPosition + (RayDirection * t);
+			//Info->Position = HitPosition;
+			Result = true;
+		}
+	}
+
+	return Result;
+}
+
 void main()
 {
 	uvec3 tid = gl_GlobalInvocationID;
 
-	uint vertid = tid.x;
+	uint vertid = tid.x * 3;
 
-	if(vertid >= meshpick.numVerts)
-		return;
+	// NOTE(james): when I turn on following early out the shader doesn't do anything.
+	// so I end up with a lot of duplicated work on smaller meshes
 
-	uint idx = meshpick.use_indices != 0u ? ib.data[vertid] : vertid;
+	//if(vertid >= meshpick.numVerts)
+	//	return;
```

Comment:
I have no idea why it doesn't return anything when I have the early out in. I spent a couple hours trying to figure it out. Not really worth spending more time on if the duplicated effort doesn't cost us too much.